src/Features/VisualBasic/Portable/ReplaceMethodWithProperty/VisualBasicReplaceMethodWithPropertyService.vb

```diff
@@ -0,0 +1,233 @@
+                        DirectCast(getAndSetMethods.GetMethodDeclaration.Parent, MethodBlockBaseSyntax).Statements)
+
+                    Dim setAccessorStatement = SyntaxFactory.SetAccessorStatement()
+                    setAccessorStatement = setAccessorStatement.WithParameterList(setMethodStatement?.ParameterList)
+
+                    If getAndSetMethods.GetMethod.DeclaredAccessibility <> getAndSetMethods.SetMethod.DeclaredAccessibility Then
+                        setAccessorStatement = DirectCast(generator.WithAccessibility(setAccessorStatement, getAndSetMethods.SetMethod.DeclaredAccessibility), AccessorStatementSyntax)
+                    End If
+
+                    Dim setAccessor = SyntaxFactory.SetAccessorBlock(setAccessorStatement,
+                        DirectCast(getAndSetMethods.SetMethodDeclaration.Parent, MethodBlockBaseSyntax).Statements)
+
+                    Dim accessors = SyntaxFactory.List({getAccessor, setAccessor})
+                    newPropertyDeclaration = SyntaxFactory.PropertyBlock(propertyStatement, accessors)
+                Else
+                    ' Methods don't have bodies.  Just make a property statement
+                    newPropertyDeclaration = propertyStatement
+                End If
+            End If
+
+            Dim trivia As IEnumerable(Of SyntaxTrivia) = getMethodStatement.GetLeadingTrivia()
+            If setMethodStatement IsNot Nothing Then
+                trivia = trivia.Concat(setMethodStatement.GetLeadingTrivia())
+            End If
+
+            newPropertyDeclaration = newPropertyDeclaration.WithLeadingTrivia(trivia)
+
+            Return newPropertyDeclaration.WithAdditionalAnnotations(Formatter.Annotation)
+        End Function
+
+        Private Function GetPropertyName(identifier As SyntaxToken, propertyName As String, nameChanged As Boolean) As SyntaxToken
+            Return If(nameChanged, SyntaxFactory.Identifier(propertyName), identifier)
+        End Function
+
+        Public Sub ReplaceGetReference(editor As SyntaxEditor, nameToken As SyntaxToken, propertyName As String, nameChanged As Boolean) Implements IReplaceMethodWithPropertyService.ReplaceGetReference
+            If nameToken.Kind() <> SyntaxKind.IdentifierToken Then
+                Return
+            End If
+
+            Dim nameNode = TryCast(nameToken.Parent, IdentifierNameSyntax)
+            If nameNode Is Nothing Then
+                Return
+            End If
+
+            Dim newName = If(nameChanged,
+                SyntaxFactory.IdentifierName(SyntaxFactory.Identifier(propertyName).WithTriviaFrom(nameToken)),
+                nameNode)
+
+            Dim parentExpression = If(nameNode.IsRightSideOfDot(), DirectCast(nameNode.Parent, ExpressionSyntax), nameNode)
+            Dim root = If(parentExpression.IsParentKind(SyntaxKind.InvocationExpression), parentExpression.Parent, parentExpression)
+
+            editor.ReplaceNode(root, parentExpression.ReplaceNode(nameNode, newName))
```

Comment:
I think it's pretty easy to do. After all, we've got the APIs for it, and we do it in most of our refactorings. It might not come up very often, but it's pretty easy to construct cases that break.

```C#
using System;

class C
{
    string GetInt32()
    {
        return "42";
    }

    void M()
    {
        var x = GetInt32();
        var s = Int32.Parse(x);
    }
}
```

Maybe just file an issue for C#/VB for now?